CHANGELOG.md

```diff
@@ -1,5 +1,301 @@
+* ask for password if -p is not provided ([#4153](https://github.com/confluentinc/ksql/pull/4153)) ([7a83bbf](https://github.com/confluentinc/ksql/commit/7a83bbf))
+* add a KSQL cache for Kafka authorization validator ([#4186](https://github.com/confluentinc/ksql/pull/4186)) ([58ed39c](https://github.com/confluentinc/ksql/commit/58ed39c))
+* add config to make error messages configurable ([#4121](https://github.com/confluentinc/ksql/pull/4121)) ([cedf47e](https://github.com/confluentinc/ksql/commit/cedf47e))
+* add COUNT_DISTINCT and allow generics in UDAFs ([#4150](https://github.com/confluentinc/ksql/pull/4150)) ([2d5e680](https://github.com/confluentinc/ksql/commit/2d5e680))
+* add source statement to SourceDescription ([#4134](https://github.com/confluentinc/ksql/pull/4134)) ([1146aa5](https://github.com/confluentinc/ksql/commit/1146aa5))
+* add support for inline struct creation ([#4120](https://github.com/confluentinc/ksql/pull/4120)) ([6e558da](https://github.com/confluentinc/ksql/commit/6e558da))
+* allow environment variables to configure embedded connect ([#4260](https://github.com/confluentinc/ksql/pull/4260)) ([e032ea9](https://github.com/confluentinc/ksql/commit/e032ea9))
+* enable Kafla ACL authorization checks for Pull Queries ([#4187](https://github.com/confluentinc/ksql/pull/4187)) ([5ee1e9e](https://github.com/confluentinc/ksql/commit/5ee1e9e))
+* implemention of KLIP-13 ([#4099](https://github.com/confluentinc/ksql/pull/4099)) ([b23dae9](https://github.com/confluentinc/ksql/commit/b23dae9))
+* show properties now includes embedded connect properties and scope ([#4099](https://github.com/confluentinc/ksql/pull/4099)) ([ebac104](https://github.com/confluentinc/ksql/commit/ebac104))
+* add connector status to LIST CONNECTORS ([#4077](https://github.com/confluentinc/ksql/pull/4077)) ([5ff94b6](https://github.com/confluentinc/ksql/commit/5ff94b6))
+* Add Cube UDTF ([#3935](https://github.com/confluentinc/ksql/pull/3935)) ([6be8e7c](https://github.com/confluentinc/ksql/commit/6be8e7c))
+* add JMX metric for commandRunner status ([#4019](https://github.com/confluentinc/ksql/pull/4019)) ([55d75f2](https://github.com/confluentinc/ksql/commit/55d75f2))
+* add support to terminate all running queries ([#3944](https://github.com/confluentinc/ksql/pull/3944)) ([abbce84](https://github.com/confluentinc/ksql/commit/abbce84))
+* expose execution plans from the ksql engine API ([#3482](https://github.com/confluentinc/ksql/pull/3482)) ([067139c](https://github.com/confluentinc/ksql/commit/067139c))
+* expression support for PARTITION BY ([#4032](https://github.com/confluentinc/ksql/pull/4032)) ([0f31f8e](https://github.com/confluentinc/ksql/commit/0f31f8e))
+* never skip command in CommandRunner thread ([#3962](https://github.com/confluentinc/ksql/pull/3962)) ([89e059b](https://github.com/confluentinc/ksql/commit/89e059b))
```

Comment:
I think that this can be removed.

Reply:
Removed from the features list, though I realize in the process that the BREAKING CHANGE in the PR description for this PR was not propagated to the actual commit body, and wasn't picked up by the changelog generator as a result (fyi @stevenpyzhang for future reference). I've manually added it now.

Reply:
Actually, I'm not sure how to describe this change -- @stevenpyzhang want to give a recommendation? 

The PR description says

> Existing command topics commands may have conflicting commands in them and could cause the server to never start up, so a new flag is introduced for transactionally produced commands to indicate that only they shouldn't be skipped since they've gone through the transactional produce protocol.

but I don't see a flag introduced in the PR.

I also can't tell how serious of a breaking change this is. If we think a significant portion of users will hit against this (have commands in their command topic that prevent them from upgrading or else those commands will be stuck, unless they set a server config), we should add a note into the upgrade notes.